test/parser/testdata.py

```diff
@@ -23,11 +29,18 @@ class ccDataTest(unittest.TestCase):
 
     def setUp(self):
         self.data = cclib.parser.logfileparser.ccData()
+        self.mock_logger = mock.Mock()
 
     def _set_attributes(self, names, val):
         for n in names:
             setattr(self.data, n, val)
 
+    def test_valuecheck_negative_etenergies(self):
+        self.data.etenergies = [1, -1]
+        self.data.check_values(logger=self.mock_logger)
+        self.mock_logger.error.assert_called_once_with(
+            AnyStringWith("At least one excitation energy is negative"))
```

Comment:
So this is similar to possible behavior using https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex, where the check is that part of the message contains `At least one excitation energy is negative`?

Reply:
Right. I didn't want to put in the whole string verbatim, and I don't think there's anything built in like this in mock.